common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -512,11 +518,23 @@ protected boolean isRequestAuthorityRealmSameAsATRealm(@NonNull final Authority
     }
 
     protected boolean isMsaAccount(final MicrosoftTokenResponse microsoftTokenResponse) {
-                final String tenantId = SchemaUtil.getTenantId(
+        final String tenantId = SchemaUtil.getTenantId(
                 microsoftTokenResponse.getClientInfo(),
                 microsoftTokenResponse.getIdToken()
         );
         return AzureActiveDirectoryAudience.MSA_MEGA_TENANT_ID.equalsIgnoreCase(tenantId);
     }
 
+    public ICacheRecord finalizeCacheRecordForResult(@NonNull final ICacheRecord cacheRecord,
+                                                     @NonNull final AbstractAuthenticationScheme scheme) throws ClientException {
+        if (scheme instanceof ITokenAuthenticationSchemeInternal) {
+            final ITokenAuthenticationSchemeInternal tokenAuthScheme = (ITokenAuthenticationSchemeInternal) scheme;
+            tokenAuthScheme.setAccessToken(cacheRecord.getAccessToken().getSecret());
+            cacheRecord
+                    .getAccessToken()
+                    .setSecret(tokenAuthScheme.getAccessTokenForAuthorizationHeader());
+        }
```

Comment:
expiry date?

Reply:
5 minutes is only a default that's configured on the _middleware_ -- I have a follow-up with our PM team as to how this should behave since the client likely shouldn't assume it always expires as 5 minutes

Reply:
(Going to leave unresolved)

Reply:
In that case, I think we could leave a TODO somewhere and call it good.

Reply:
@rpdome I'll put a TODO in MSAL's `AuthenticationResult`

Reply:
Added here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/837/commits/f53f05b733d9d79f144ce58e995cdcc9348ae168